crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -369,9 +367,8 @@ def proposal_update
       return render :text => ret[1], :status => ret[0] if ret[0] != 200
       return render :json => ret[1]
     else # This is UI.
-      params[:id] = "bc-#{params[:barclamp]}-#{params[:id] || params[:name]}"
       if params[:submit] == t('barclamp.proposal_show.save_proposal')
-        @proposal = ProposalObject.find_proposal_by_id(params[:id])
+        @proposal = Proposal.where(barclamp: params[:barclamp], name: params[:id] || params[:name]).first
```

Comment:
Looking at how the `params[:id]` is constructed on line [372](https://github.com/crowbar/barclamp-crowbar/pull/1264/files#diff-6e0801372d9a358289edc702ef124633L372) - it seems that this should continue working as expected?